test/compress/evaluate.js

```diff
@@ -1306,3 +1306,29 @@ issue_2535_2: {
         x();
     }
 }
+
+issue_2535_3: {
+    options = {
+        booleans: true,
+        evaluate: true,
+    }
+    input: {
+        console.log(Object(1) && 1 && 2);
+        console.log(2==Object(1) || 0 || void 0 || null);
+        console.log(Object(1) && true && 1 && 2 && Object(2));
+        console.log(2==Object(1) || 0 || void 0 || null || Object(2));
```

Comment:
Please add:
```js
console.log(Object(1) && true && 1 && null && 2 && Object(2));
console.log(2==Object(1) || 0 || void 0 || "ok" || null || Object(2));
```

Reply:
Done in https://github.com/mishoo/UglifyJS2/pull/2551/commits/559c4289c838e5153d31938b042b0ec0ee1dc491

Reply:
thanks!